packages/@uppy/companion-client/src/Provider.ts

```diff
@@ -120,32 +120,32 @@ export default class Provider<
 
   onReceiveResponse(response: Response): Response {
     super.onReceiveResponse(response)
-    const plugin = this.uppy.getPlugin(this.pluginId)
-    const oldAuthenticated = plugin!.getPluginState().authenticated
+    const plugin = this.getPlugin()
+    const oldAuthenticated = plugin.getPluginState().authenticated
     const authenticated = oldAuthenticated
       ? response.status !== authErrorStatusCode
       : response.status < 400
-    plugin!.setPluginState({ authenticated })
+    plugin.setPluginState({ authenticated })
     return response
   }
 
   async setAuthToken(token: string): Promise<void> {
-    return (
-      this.uppy
-        .getPlugin(this.pluginId)!
-        // @ts-expect-error don't think we can make core aware that _some_ plugins have a storage
-        .storage.setItem(this.tokenKey, token)
-    )
+    return this.getPlugin().storage.setItem(this.tokenKey, token)
   }
 
-  async #getAuthToken(): Promise<string | void> {
-    // @ts-expect-error don't think we can make core aware that _some_ plugins have a storage
-    return this.uppy.getPlugin(this.pluginId).storage.getItem(this.tokenKey)
+  async #getAuthToken(): Promise<string | null> {
+    return this.getPlugin().storage.getItem(this.tokenKey)
   }
 
   protected async removeAuthToken(): Promise<void> {
-    // @ts-expect-error don't think we can make core aware that _some_ plugins have a storage
-    return this.uppy.getPlugin(this.pluginId).storage.removeItem(this.tokenKey)
+    return this.getPlugin().storage.removeItem(this.tokenKey)
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  protected getPlugin() {
```

Comment:
```suggestion
  #getPlugin() {
```

Reply:
I thought we don't want to change the source?

Reply:
this method was added by me. i don't remember why i made it protected, but I agree it should be private instead

Reply:
AFAICT this method is brand new, at least that's what I get from the output diff 🤔